### PR TITLE
Replace broken Eigen3 with correctly spelled EIGEN3

### PR DESCRIPTION
**Pull request by davetcoleman**
Per https://github.com/ros-planning/moveit/pull/252#discussion_r80627429

### CODE REVIEW

**Comment by gavanderhoorn** [comment]
But let's wait on Travis to agree.

**Comment by v4hn** [comment]
Please also remove the explicit `EIGEN3_INCLUDE_DIRS` you added in #252 .
DEPENDS automagically takes care of this.

Looking into the details here, `grep -r include.*Eigen moveit_experimental` does not return any results, so either someone forgot an include or the package should not actually depend on Eigen3 at all.

Looking even further `moveit_experimental/kinematics_constraint_aware/src/kinematics_constraint_aware.cpp` uses the Eigen::Affine3d datatype but does not include `Eigen/Geometry`. So the include is missing there and thus the package *does* depend on eigen3...

**Comment by davetcoleman** [comment]
> Please also remove the explicit EIGEN3_INCLUDE_DIRS

I copied that from [moveit_core](https://github.com/ros-planning/moveit/blob/kinetic-devel/moveit_core/CMakeLists.txt#L85). Assuming you are correct about not needing the include, I've made the change throughout all the moveit packages in this PR.

> So the include is missing there and thus the package does depend on eigen3...

I just added the include for this PR also

**Comment by v4hn** [comment]
On Tue, Sep 27, 2016 at 08:48:31AM -0700, Dave Coleman wrote:
> Assuming you are correct about not needing the include, I've made the change throughout all the moveit packages in this PR.

I am correct, but you probably want to wait for https://github.com/ros/eigen_stl_containers/pull/6 . I noticed the bug there when I catkin_lint'ed `moveit_core`.
The request has basically been open and unanswered for 3 months now until dirk took an interest two days ago because of another bug report.
So hopefully this will be fixed soon, and yes, at that point we should be able to clean that up throughout MoveIt :)

**Comment by davetcoleman** [comment]
https://github.com/ros/eigen_stl_containers/pull/6 was just merged. So this PR can be merged now?

Also are you not assuming that all our packages that use Eigen also use eigen_stl_containers? I believe this is not true.

File: moveit_ros/visualization/CMakeLists.txt [hunk 1]
```diff
@@ -65,11 +65,11 @@ catkin_package(
     planning_scene_rviz_plugin/include
     motion_planning_rviz_plugin/include
     trajectory_rviz_plugin/include
-    ${EIGEN3_INCLUDE_DIRS}
   CATKIN_DEPENDS
+    EIGEN3
```
Comment (v4hn) [inline moveit_ros/visualization/CMakeLists.txt]: this is wrong. `EIGEN3` has to go into a `DEPENDS` section
Same for most of the other places you changed.

**Comment by v4hn** [comment]
> ros/eigen_stl_containers#6 was just merged. So this PR can be merged now?
> 
> Also are you not assuming that all our packages that use Eigen also use eigen_stl_containers? I believe this is not true.

No, but I remember hitting that bug when I updated a package (not `moveit_core` actually) that relied on `eigen_stl_containers` but did not include Eigen directly. Don't remember which module it was right now.

**Comment by gavanderhoorn** [comment]
Not targeted at @davetcoleman, but is there a way to revoke your PR 'approved' status? Seems strange that you can approve a PR, then the contributor submits more changes, but the approval status is not reset/changed.

File: moveit_ros/visualization/CMakeLists.txt (hunk 1 above)
Comment (davetcoleman) [inline moveit_ros/visualization/CMakeLists.txt]: fixed

**Comment by davetcoleman** [comment]
Travis passes shadow-fixed (see https://github.com/ros-planning/moveit/pull/257) and I have addressed the feedback. Ready to merge.

**Comment by davetcoleman** [comment]
ping

**Comment by davetcoleman** [comment]
@v4hn I believe I addressed all your feedback, could you approve this PR and merge it?

**Comment by v4hn** [comment]
Finally, I'm somewhat back to the keyboard. Sorry for the long timeout.

@k-okada pointed out that Wily (supported in kinetic) does not have eigen3.3 (which is about to be released this month after more than a year of soak time :) ...).
Before Eigen3.3 `find_package(Eigen3)` did only define `${EIGEN3_INCLUDE_DIR}` and not `${EIGEN3_INCLUDE_DIRS}`.

So even in kinetic, we are required to have this workaround after each eigen-find_package:
> if(NOT EIGEN3_INCLUDE_DIRS)
>  set(EIGEN3_INCLUDE_DIRS ${EIGEN3_INCLUDE_DIR})
> endif()

https://github.com/ros/eigen_stl_containers/pull/6/commits/2db0ce7e1dfd141974b7a3b7f95fbce66f4e71e2

Could you please add this to the request?

**Comment by davetcoleman** [comment]
I've added the INCLUDE_DIRS fix, build passes

**Pull request merged**